modules/collabora_online_group/tests/src/Functional/GroupMediaViewsTest.php

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * Test the modifications performend in groupmedia view by the module.
+ */
+class GroupMediaViewsTest extends BrowserTestBase {
+
+  use MediaCreationTrait;
+  use GroupRelationTrait;
+  use GroupTestTrait;
+  use MediaTypeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online_group',
+    'views',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Test the links for Collabora operations in the view.
+   */
+  public function testViewLinks(): void {
+    // Add configuration needed for testing.
+    $group_type = $this->createGroupType(['id' => 'group_type_1']);
+    $media_type = $this->createMediaType('file', ['id' => 'document']);
+    $this->createGroupRole([
+      'group_type' => 'group_type_1',
+      'scope' => PermissionScopeInterface::INSIDER_ID,
+      'global_role' => RoleInterface::AUTHENTICATED_ID,
+      'permissions' => [
+        'view group',
+        'access group_media overview',
+        'view group_media:document entity',
+        'edit any group_media:document in collabora',
+        'preview group_media:document in collabora',
+      ],
+    ]);
+    $this->createPluginRelation($group_type, 'group_media:document', [
+      'group_cardinality' => 0,
+      'entity_cardinality' => 1,
+      'use_creation_wizard' => FALSE,
+    ]);
+
+    // Create content.
+    $group = $this->createGroup(['type' => 'group_type_1']);
+    for ($i = 1; $i < 4; $i++) {
+      $media = $this->createMediaEntity('document', [
+        'id' => 'media_' . $i,
+        'name' => 'Media ' . $i,
+      ]);
+      $group->addRelationship($media, 'group_media:document');
+    }
+    $user = $this->createUser([
+      'view the administration theme',
+      'access administration pages',
+      'access group overview',
+    ]);
+    $group->addMember($user);
+
+    // Go to the page and check the links added to the view.
+    $this->drupalLogin($user);
+    $this->drupalGet("group/{$group->id()}/media");
+    $assert_session = $this->assertSession();
+
+    // Check table header.
+    $table = $assert_session->elementExists('css', 'table');
+    $table_header = $assert_session->elementExists('css', 'thead', $table);
+    $rows = $table_header->findAll('css', 'tr');
+    $cols = $rows[0]->findAll('css', 'th');
+    $this->assertEquals('Media name', $cols[0]->getText());
+    $this->assertEquals('Bundle', $cols[1]->getText());
+    $this->assertEquals('Status', $cols[2]->getText());
+    $this->assertEquals('Publisher', $cols[3]->getText());
+    // Support for different vesrions of groupmedia.
+    $this->assertTrue(in_array($cols[4]->getText(), ['Operations', 'Dropbutton']));
+
+    // Check that rows contain new links for operations in Collabora.
+    $table_body = $assert_session->elementExists('css', 'tbody', $table);
+    $rows = $table_body->findAll('css', 'tr');
+    $i = 0;
+    foreach (Media::loadMultiple() as $media) {
```

Comment:
I thought I had somewhere a review comment suggesting this:
```suggestion
    $media_entities = Media::loadMultiple();
    foreach (array_values($media_entities) as $i => $media) {
```
or
```suggestion
    foreach (array_values(Media::loadMultiple()) as $i => $media) {
```
But, let's keep it as is for now.
I just wanted to mention the option.